heat/core/tests/test_suites/basic_test.py

```diff
@@ -0,0 +1,173 @@
+from unittest import TestCase
+from heat.core import dndarray, MPICommunication, MPI, types, factories
+import heat as ht
+
+import numpy as np
+import torch
+
+
+class BasicTest(TestCase):
+
+    __comm = MPICommunication()
+    __device = None
+
+    @property
+    def comm(self):
+        return BasicTest.__comm
+
+    @property
+    def device(self):
+        return BasicTest.__device
+
+    def get_rank(self):
+        return self.comm.rank
+
+    def get_size(self):
+        return self.comm.size
+
+    def assert_array_equal(self, heat_array: dndarray.DNDarray, numpy_array):
+        """
+        Check if the heat_array is equivalent to the numpy_array. Therefore first the split heat_array is compared to
+        the corresponding numpy_array slice locally and second the heat_array is combined and fully compared with the
+        numpy_array.
+        Note if the heat array is split it also needs to be balanced.
```

Comment:
a secondary note should be added here that the standard numpy dtype is float64. this is different from torch/heat where the default if float32.  essentially, the numpy arrays used in the function should be created with the proper dtype

Reply:
Is this a general rule or can there be changes or single functions that differ in their behaviour? If the user wants to be certain, he can always pass args where the dtype is explicitly defined. I just don't want to make any general assumptions about the behaviour of the functions which might not be correct and result in unexpected results.

Reply:
yes the dtype can change after a function. my concern is that the types will not match up because of the difference in what is the standard types of the packages. 
i think that the thing to do would be to create the numpy array with the dtype of the heat array, and if no dtype is available then it should be created with dtype=float32. that way they have the same starting point. and i believe that numpy will only increase the precision if it is necessary. i.e. type changes which occur in the course of the function will remain at the same precision.